code/modules/multiz/movement/turf/turf_zfall.dm

```diff
@@ -53,17 +53,25 @@
 	if(!can_start_zFall(A, target, force, from_zfall))
 		return FALSE
 	if(from_zfall) // if this is a >1 level fall
-		sleep(2) // This is the delay between falling zlevels. Otherwise zfalls would be instant to the client, which does not look great.
-		var/turf/new_turf = get_turf(A) // make sure we didn't move onto a solid turf, if we did this will perform a zimpact via the caller
-		target = get_step_multiz(new_turf, DOWN)
-		if(!new_turf.can_start_zFall(A, target, force, from_zfall))
-			new_turf.do_z_impact(A, levels - 1)
-			return TRUE // skip parent zimpact - do a zimpact on new turf, the turf below us is solid
```

Comment:
these returns being removed, does this impact any caller behavior?

Reply:
Wait, if both conclusions to Zfall finish are true, cant I just add a line after the zfall timer returning true?

Reply:
I'm not sure if this is logically equivalent but it sort of seems like it would work so sure